x/upgrade/keeper/keeper.go

```diff
@@ -48,6 +50,40 @@ func (k Keeper) SetUpgradeHandler(name string, upgradeHandler types.UpgradeHandl
 	k.upgradeHandlers[name] = upgradeHandler
 }
 
+func (k *Keeper) SetVersionManager(vm module.VersionManager) {
+	k.versionManager = vm
+}
+
+// SetConsensusVersions saves the consensus versions retrieved from module.Manager
+func (k Keeper) SetConsensusVersions(ctx sdk.Context) {
+	modules := k.versionManager.GetConsensusVersions()
+	store := ctx.KVStore(k.storeKey)
+	migrationStore := prefix.NewStore(store, []byte{types.MigrationMapByte})
+	for modName, ver := range modules {
+		nameBytes := []byte(modName)
+		verBytes := make([]byte, 8)
+		binary.LittleEndian.PutUint64(verBytes, ver)
+		migrationStore.Set(nameBytes, verBytes)
+	}
+}
+
+// GetConsensusVersions gets a MigrationMap from state
+func (k Keeper) GetConsensusVersions(ctx sdk.Context) module.MigrationMap {
+	store := ctx.KVStore(k.storeKey)
+	it := sdk.KVStorePrefixIterator(store, []byte{types.MigrationMapByte})
+
+	migmap := make(module.MigrationMap)
+	defer it.Close()
+	for ; it.Valid(); it.Next() {
+		moduleBytes := it.Key()
+		name := string(moduleBytes[1:])
```

Comment:
```suggestion
		name := string(moduleBytes)
```
i think it should be like this, because the iterator is already on a prefixed store.

Reply:
False alert, it's correct with the [1:]

Reply:
A comment would be good here

Reply:
[comment added](https://github.com/cosmos/cosmos-sdk/pull/8743/files#diff-33ff09992f1a3583e2aa8eb932c62c833d52e446488070ed958eb672eed8da3eR74)